frontend/src/features/admin-form/create/logic/components/LogicContent/EditLogicBlock/EditCondition/ThenShowBlock.tsx

```diff
@@ -70,13 +74,67 @@ export const ThenShowBlock = ({
       : 'show'
   }, [logicTypeValue])
 
+  const [deletedFieldsCount, setDeletedFieldsCount] = useState<number>(0)
+
+  /**
+   * Compute whether any/all fields in the show fields are deleted, then run
+   * effect to delete fields on render and show appropriate error/infobox.
+   */
+  const showValueWatch = useWatchDependency(watch, 'show')
+
+  useEffect(() => {
+    if (
+      logicTypeValue !== LogicType.ShowFields ||
+      !showValueWatch.value ||
+      !mapIdToField
+    )
+      return
+    const filteredShowFields = showValueWatch.value.filter(
+      (field) => field in mapIdToField,
+    )
+    const deletedFieldsCount =
```

Comment:
can move this line below, closer to where it's used

Reply:
ok!